src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs

```diff
@@ -260,10 +245,10 @@ internal static SyntaxToken RescanInterpolatedString(InterpolatedStringExpressio
                 interpolatedString.GetLastToken().GetTrailingTrivia());
         }
 
-        private class InterpolatedStringScanner
+        private struct InterpolatedStringScanner
```

Comment:
this was made a struct as it is not copied and only sits on one stack frame.

Reply:
Just curious, what enforces "NonCopyable"?

Reply:
We have an analyzer that does it :)